Fullsync path logging cleanup

bsparrow435 opened the pull request:
This branch addresses some of the logging issues we've had while testing aae fullsync. Specifically log spam and incorrectly classified log messages. All changes in this branch are to be logging and no functional changes.

bsparrow435 committed:
Logging cleanup

cmeiklejohn commented on src/riak_core_connection_mgr.erl:
```diff
@@ -520,7 +515,7 @@ locate_endpoints({Type, Name}, Strategy, Locators) ->
             end;
         error ->
-            lager:info("Unknown target: ~p ~p", [Name, Strategy]),
+            lager:error("Unknown target: ~p ~p", [Name, Strategy]),
```
Are we sure about this change?  If a user sees this it might be confusing as it contains very little information.

bsparrow435 replied:
This is not a code path I have ever hit either in development or in a customers logs so I am not 100% sure of the correct log level. Since i've never hit the path and it is a match on `error` i figured error was a more appropriate level. Maybe `warning` is better.

cmeiklejohn commented on src/riak_repl2_fscoordinator.erl:
```diff
@@ -255,12 +255,12 @@ handle_call({node_clean, Node}, _From, State = #state{dirty_nodes=DirtyNodes}) -
     {reply, ok, NewState};
 
 handle_call(_Request, _From, State) ->
-    lager:info("ignoring ~p", [_Request]),
```
Why is this not debug level?

bsparrow435 replied:
For all of the modules changed in this PR if I saw an `info` level log for a catch-all handle_call, handle_cast, handle_etc, I changed it to `notice`. My reasoning was to not alarm users but also make them visible in default log configurations. Is there a standard way we handle these catch-all's in other projects that's better? Maybe we should just delete this function and let erlang be erlang.

cmeiklejohn commented:
Are we sure about removing the anya lines?  Not sure how valuable this is to debugging.  cc: @jonmeredith

cmeiklejohn commented:
I'm mostly :+1: on this, however, we should cut down whatever logging we don't absolutely need because we did determine during aae benchmarking that it is introducing overhead, even for lager:debug calls.

cmeiklejohn commented:
Please be sure that this makes it into the kill bill so it's accounted for and we don't forget to merge it.

jonmeredith commented:
Debug level is fine for me...

On Sat, Mar 1, 2014 at 6:00 PM, Christopher Meiklejohn <
notifications@github.com> wrote:

> Are we sure about removing the anya lines? Not sure how valuable this is
> to debugging. cc: @jonmeredith https://github.com/jonmeredith
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/basho/riak_repl/pull/549#issuecomment-36442375
> .

## 

Jon Meredith
VP, Engineering
Basho Technologies, Inc.
jmeredith@basho.com

cmeiklejohn commented:
What's the deal with this?  Is this targeted for 2.0?

bsparrow435 commented:
Sure! I owe one more run through to eliminate further, not needed, debug logs. I'll do it later today then I think we're good for merge after a rebase.

bsparrow435 committed:
Address PR comments

Removed more unneeded log entries. Most are available in other log
messages later in the fullsync(such as the NDiff for the ebloom built
with AAE differences). Others are removing unnecessary logs from
catch-all handle_* functions.

bsparrow435 commented:
@cmeiklejohn this should be ready for review

cmeiklejohn commented:
Can you rebase/squash this into one commit?

andrewjstone commented:
- [ ] code review
- [ ] replication2

andrewjstone commented on src/riak_repl2_fscoordinator.erl:
```diff
@@ -331,14 +330,13 @@ handle_cast(stop_fullsync, State) ->
     {noreply, State2};
 
 handle_cast(_Msg, State) ->
-    lager:info("ignoring ~p", [_Msg]),
```
I'm not sure removing this is a good idea, unless we expect useless messages. In my experience that usually indicates some bug and it'd be nice to keep it around. But I'm not sure how relevant that is in this code.

andrewjstone commented on src/riak_repl2_fscoordinator.erl:
```diff
@@ -391,7 +389,7 @@ handle_info({'EXIT', Pid, Cause},
                 N when N > RetryLimit, is_integer(RetryLimit) ->
-                    lager:warning("fssource dropping partition: ~p, ~p failed"
+                    lager:warning("Fullsync dropping partition: ~p, ~p failed"
                                "retries", [Partition, RetryLimit]),
```
Is this valid lagering? The "retries" as second parameter that is.

seancribbs replied:
Actually that will appear as "failedretries", successive 'strings' are concatenated by the compiler.

cmeiklejohn replied:
Please fix.

andrewjstone commented on src/riak_repl2_fscoordinator.erl:
```diff
@@ -462,7 +460,6 @@ handle_info(send_next_whereis_req, State) ->
     {noreply, State2};
 
 handle_info(_Info, State) ->
-    lager:info("ignoring ~p", [_Info]),
```
I say let it crash or log it. just my 2c.

cmeiklejohn replied:
Logging is fine, but definitely not at info level.

andrewjstone commented on src/riak_repl2_fscoordinator.erl:
```diff
@@ -695,11 +692,11 @@ start_fssource(Partition2={Partition,_,_} = PartitionVal, Ip, Port, State) ->
                 {already_started, OtherPid} ->
-                    lager:notice("A fullsync for partition ~p is already in"
+                    lager:info("A fullsync for partition ~p is already in"
                         " progress for ~p", [Partition,
```
Is this a valid 3 param lager call?

andrewjstone commented on src/riak_repl2_fscoordinator.erl:
```diff
@@ -695,11 +692,11 @@ start_fssource(Partition2={Partition,_,_} = PartitionVal, Ip, Port, State) ->
                             riak_repl2_fssource:cluster_name(OtherPid)]);
                 {{max_concurrency, Lock},_ChildSpec} ->
-                    lager:notice("Fullsync for partition ~p postponed"
+                    lager:debug("Fullsync for partition ~p postponed"
```
I'm starting to believe these are all valid. :)

cmeiklejohn commented:
Needs comments addressed and clean rebase/squash against master before merge.

cmeiklejohn commented:
Closing.
